### DISCUSSION TITLE
[Android] OnCreate method in Activity doesn't get called in Release configuration.

### DISCUSSION THREAD

**Discussion by lukaszossowski** [AvaloniaUI/Avalonia#8989] [Q&A] [answered]
I have already created an issue but there was no response. I am writing my app using the latest 11 preview version of Avalonia in VS2022 and everything works fine when I am in debug configuration. However, when I try to run the app in Release mode it just gets stuck on splash screen.

I tried poor man's debugging with vibrations:
```
 protected override void OnCreate(Bundle savedInstanceState)
 {
     Vibration.Vibrate();
     base.OnCreate(savedInstanceState);
 }
```
And it turns out that it never gets called. It does work in debug.

To eliminate any variables I created a new project using the avalonia cross plaform app template and even the clean project has the same issues - works in debug, but not in release.

Do you have any advice on how to deal with it?

**Comment by maxkatz6** [comment]
Can you try to test the same without avalonia, but with simple .net 6 android template?

**Reply by maxkatz6** [reply]
Also I wonder if aot and trimming dotnet project options can affect it.

**Reply by maxkatz6** [reply]
Also normally it works with new avalonia template and control catalog, when built in release and run on real device.  Could be device specific.

**Reply by lukaszossowski** [reply]
I just made a .NET android app and it works in both debug and release mode. I have also given the .apk of release mode of my app to my friend and it also gets stuck in the same place.

I also just created a new avalonia cross-platform project from provided templates and it gets stuck as well. Everything using the 11 preview1 version.

I have a Galaxy S20 Ultra phone.

**Reply by maxkatz6** [reply]
Can you build an avalonia apk and sent here? Just want to try on my device.

And don't have access to my PC yet, so I can't build with latest avalonia yet by myself.

**Comment by lukaszossowski** [comment]
Sure, Here is the 

> com.CompanyName.Project5-Signed.apk

https://drive.google.com/file/d/1NsmfxH5jXWohl10yXxholMq3gtfC3_UG/view?usp=sharing

that was created straight from the template.

**Reply by maxkatz6** [reply]
Can confirm. Splash screen is never ending on my OnePlus either.

**Comment by lukaszossowski** [comment, marked as answer]
Is there any update on this? The preview 2 version of Avalonia doesn't seem to change anything, result is the same. I tried using logcat to see what is happening but there was only one thing that caught my attention as it seems to be happening constantly (filtered by transit_activity_open):
![logcat](https://user-images.githubusercontent.com/112421929/195715022-96a79519-d505-4354-be16-a35fb2245171.JPG)

The warning part says this:
`transition: mNextAppTransition=TRANSIT_ACTIVITY_OPEN, displayId: 0 Callers=com.android.server.wm.RootWindowContainer.executeAppTransitionForAllDisplay:2586 com.android.server.wm.ActivityStackSupervisor.reportResumedActivityLocked:2265 com.android.server.wm.ActivityRecord.completeResumeLocked:5705 com.android.server.wm.ActivityStack.minimalResumeActivityLocked:1225 com.android.server.wm.ActivityStackSupervisor.realStartActivityLocked:1152  `

I'd really like to be able to do the release build and use AOT since it seems to be making my app work much smoother, but I can only test that using a debug build with fast deployment (otherwise there are a lot of errors in some java code).

EDIT: It seems that disabling Startup Tracing fixes the problem, as my app launches fine without it.
